### PR TITLE
CI: Pin pyarrow smaller than 10

### PR DESCRIPTION
**Pull request by phofl**
lets see if this fixes ci, matplotlib is a bit weird

### CODE REVIEW

File: ci/deps/actions-310.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ dependencies:
   - psycopg2
   - pymysql
   - pytables
-  - pyarrow
+  - pyarrow<10
```
Comment (MarcoGorelli) [inline ci/deps/actions-310.yaml:45]: the syntax doesn't seem to work, how about `<10.0.0`?

**Comment by MarcoGorelli** [comment]
>  matplotlib is a bit weird

🤔 yeah, why is version 3.3.2 getting installed?

**Review by MarcoGorelli — APPROVED** [review]
This seems to have solved the matplotlib issue, let's merge green-ish / soon-ish

**Comment by MarcoGorelli** [comment]
merging to fix CI, thanks @phofl !

**Comment by phofl** [comment]
3.6.2 was pulled on main, same version as before but plotting tests were failing anyway

**Comment by MarcoGorelli** [comment]
are you sure? [this build from main](https://github.com/pandas-dev/pandas/actions/runs/3614121669/jobs/6090344581) had matplotlib 3.3.2, which was causing the plotting tests to fail

**Comment by phofl** [comment]
Guess: A dependency of a dependency pinned arrow recently and we pulled a really old build there that did not pin arrow but matplotlib.

**Comment by MarcoGorelli** [comment]
this needs backporting, 1.5.x CI is failing

**Comment by MarcoGorelli** [comment]
@meeseeksmachine backport 1.5.x

**Pull request merged**